src/encoder.js

```diff
@@ -4,10 +4,65 @@ const BigNumber = require('bignumber.js');
 const bs58 = require('bs58');
 
 const Utils = require('./utils');
-
-const PADDED_BYTES = 64;
+const Constants = require('./constants');
 
 class Encoder {
+  /*
+   * Encodes a parameter to hex based on its type.
+   * @param type {String} The type of the value.
+   * @param value {String|Bool|Number|HexNumber|BigNumber|BN} Value to convert to hex.
+   * @return The value converted to hex string.
+   */
+  static encodeParam(type, value) {
+    let hex = '';
+    if (type.match(Constants.ADDRESS)) {
+      if (value instanceof Array) {
+        _.each(value, (addr) => {
+          hex += this.addressToHex(addr);
+        });
+      } else {
+        hex = this.addressToHex(value);
+      }
+    } else if (type.match(Constants.BOOL)) {
+      if (value instanceof Array) {
+        _.each(value, (bool) => {
+          hex += this.boolToHex(bool);
+        });
+      } else {
+        hex = this.boolToHex(value);
+      }
+    } else if (type.match(Constants.REGEX_INT)) { // match order matters here, match int before uint
+      if (value instanceof Array) {
+        _.each(value, (int) => {
+          hex += this.intToHex(int);
+        });
+      } else {
+        hex = this.intToHex(value);
+      }
+    } else if (type.match(Constants.REGEX_UINT)) {
+      if (value instanceof Array) {
+        _.each(value, (uint) => {
+          hex += this.uintToHex(uint);
+        });
+      } else {
+        hex = this.uintToHex(value);
+      }
+    } else if (type.match(Constants.REGEX_BYTES)) { // fixed bytes, ie. bytes32
+      hex = this.stringToHex(value, Constants.MAX_HEX_CHARS_PER_BYTE);
+    } else if (type.match(Constants.REGEX_STATIC_BYTES_ARRAY)) { // fixed bytes array, ie. bytes32[10]
+      const arrCapacity = _.toNumber(type.match(Constants.REGEX_NUMBER)[1]);
+      if (value instanceof Array) {
+        hex = this.stringArrayToHex(value, arrCapacity);
+      } else {
+        hex = this.stringToHex(value, Constants.MAX_HEX_CHARS_PER_BYTE * arrCapacity);
+      }
+    } else {
+      console.error(`Unimplemented type: ${type}`);
```

Comment:
so we dunt support encoding of dynamic array?

Reply:
We do. L129 of `contract.js` uses `Encoder.encodeParam` method. When we send it from contract.js, we are encoding it as the array type individually. For code reusability.